### PR TITLE
sys: util: Add rounded div macro

### PR DESCRIPTION
**Pull request by omkar3141**
sys: util: Add rounded div macro

Signed-off-by: Omkar Kulkarni <omkar.kulkarni@nordicsemi.no>

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: include/sys/util.h [hunk 1]
```diff
@@ -131,6 +131,11 @@ extern "C" {
 #define CONTAINER_OF(ptr, type, field) \
 	((type *)(((char *)(ptr)) - offsetof(type, field)))
 
+/**
+ * @brief Returns rounded division of @p A divided by @p B.
+ */
+#define ROUNDED_DIV(A, B) (((A) + ((B) / 2)) / (B))
```
Comment (maxbachmann) [inline include/sys/util.h:137]: This evaluates the argument B twice. Therefore I think this should be a static inline function, or have a similar note to `CLAMP`.

**Review by andyross — COMMENTED** [review]
These seems a little obscure, is this really something we want in a public API?  If this was a cleanup where you found a half dozen users of the trick and unified them, that would be one thing.  But... this is just a bare API with no users.  Seems premature.  Maybe submit with the application you're thinking of?

**Comment by omkar3141** [comment]
> These seems a little obscure, is this really something we want in a public API? If this was a cleanup where you found a half dozen users of the trick and unified them, that would be one thing. But... this is just a bare API with no users. Seems premature. Maybe submit with the application you're thinking of?

This will be used here:
https://github.com/nrfconnect/sdk-nrf/pull/4191

This PR is created for upstream as per the suggestion of @trond-snekvik .

**Comment by andyross** [comment]
Meh.  Not worth a -1 I guess, but...  We're an OS, not a utility library or a dumping ground for experimental API paradigms.  If we provide an API, IMHO it ought to be because we actually **need** in multiple places in the source tree and can't get it elsewhere in a standard/portable/clean way.  Good examples are ARRAY_SIZE(), ROUND_UP/DOWN(), MIN/MAX(), CONTAINER_OF(), etc...

I just don't see how this qualifies if nothing in tree right now cares, and the only thing that does is a unmerged pull request to an out of tree module.

**Comment by omkar3141** [comment]
Ok, I understand. 

@trond-snekvik perhaps we must include this downstream then?

**Pull request closed without merging**